### ISSUE TITLE
wrong typescript definition

### ISSUE THREAD

**Issue by apptaro** [pixijs/pixi.js#6406]
Some of the typescript definition is wrong in 5.2.1.
Below is the ones I found. The first one is the current definition, the second one is the correct definition.

```
PIXI.Application.destroy(removeView?: boolean, stageOptions?: { children?: boolean; texture?: boolean; baseTexture?: boolean; }): void
PIXI.Application.destroy(removeView?: boolean, stageOptions?: boolean | { children?: boolean; texture?: boolean; baseTexture?: boolean; }): void

PIXI.Graphics.destroy(options?: { children?: boolean; texture?: boolean; baseTexture?: boolean; }): void
PIXI.Graphics.destroy(options?: boolean | { children?: boolean; texture?: boolean; baseTexture?: boolean; }): void

PIXI.Sprite.destroy(options?: { children?: boolean; texture?: boolean; baseTexture?: boolean; }): void
PIXI.Sprite.destroy(options?: boolean | { children?: boolean; texture?: boolean; baseTexture?: boolean; }): void

PIXI.Bounds.getRectangle(rect: PIXI.Rectangle): PIXI.Rectangle
PIXI.Bounds.getRectangle(rect?: PIXI.Rectangle): PIXI.Rectangle

```

**Comment by eXponenta** [comment]
Second

**Comment by ivanpopelyshev** [comment]
Our jsdoc->ts.d converter fails on `{object|boolean}` type, thats the problem with `destroy()`

`getRectangle` - that can be fixed easily. Do you want to make a PR?

**Comment by laino** [comment]
Same issue as https://github.com/pixijs/pixi.js/issues/5978

**Comment by laino** [comment]
Still a bug and the other issue got auto-closed.

**Comment by laino** [comment]
Bump. Not sure if this is still a thing.

**Comment by borodean** [comment]
Still an issue.

**Comment by bigtimebuddy** [comment]
Should be fixed in 5.4.0 prerelease, if it’s not please file a new issue with the specific type errors. We changed the mechanism for generating types and believe the bugs that were causing these issues is not there anymore.
